templates/micronaut/files/.platform.app.yaml

```diff
@@ -0,0 +1,31 @@
+# This file describes an application. You can have multiple applications
+# in the same project.
+#
+# See https://docs.platform.sh/user_guide/reference/platform-app-yaml.html
+
+# The name of this app. Must be unique within a project.
+name: app
+
+# The runtime the application uses.
+type: "java:8"
+
+disk: 1024
+
+# The hooks executed at various points in the lifecycle of the application.
+hooks:
+    build: |
+      mvn clean install -DskipTests
+
+# The relationships of the application with services or other applications.
+#
+# The left-hand side is the name of the relationship as it will be exposed
+# to the application in the PLATFORM_RELATIONSHIPS variable. The right-hand
+# side is in the form `<service name>:<endpoint name>`.
+#relationships:
+#    database: "database:mysql"
+
+# The configuration of app when it is exposed to the web.
+web:
+    commands:
+        start: |
+          java -Xmx512m -jar target/micronaut-1.0-SNAPSHOT.jar
```

Comment:
Specifying the memory cap makes me uneasy, as if you're on a much larger container this will mean you basically never use it.  However, setting it too high will cause a crash on dev instances.

I'm not sure what to do with it here, but we really need to talk to Engineering about making this value safely dynamic.

Reply:
Yes, I totally agree with you, we had this discussion on engineer Chanel, however, while they don't fix it I don't see another option.